core/src/main/scala/org/apache/spark/util/io/ChunkedByteBuffer.scala

```diff
@@ -40,6 +40,8 @@ private[spark] class ChunkedByteBuffer(var chunks: Array[ByteBuffer]) {
   require(chunks != null, "chunks must not be null")
   require(chunks.forall(_.position() == 0), "chunks' positions must be 0")
 
+  private val NIO_BUFFER_LIMIT = 64 * 1024 * 1024 // Chunk size in bytes
```

Comment:
shall we make it configurable? we can add a new entry in `org.apache.spark.internal.config`, and mark it as an internal config first.